### PR TITLE
UDP datagram packet connection support.

### PR DESCRIPTION
**Pull request by tswindell**
Signed-off-by: Tom Swindell <t.swindell@rubyx.co.uk>

### CODE REVIEW

File: net.go [hunk 1]
```diff
@@ -242,6 +242,63 @@ func WrapNetListener(nl net.Listener) (Listener, error) {
...
+        return nil, err
+    }
+
+    pc, err := net.ListenPacket(lnet, lnaddr)
+    if err != nil {
+        return nil, err
+    }
+
+	// We want to fetch the new multiaddr from the listener, as it may
+	// have resolved to some other value. WrapPacketConn does this.
```
Comment (lgierth) [inline net.go]: Should run gofmt, your code is all space-indented

Comment (tswindell) [inline net.go]: Cool, thanks I'll update and write tests shortly.

**Comment by tswindell** [comment]
Okay, added test and gofmt'd code.

**Comment by hsanjuan** [comment]
@lgierth wanna follow up here?

**Comment by lgierth** [comment]
Oh man this totally fell of my radar, sorry -- thanks so much @tswindell! I've rebased this on master in #31 and merged that.

**Pull request closed without merging**